src/main/java/fi/otavanopisto/kuntaapi/server/integrations/linkedevents/LinkedEventsEventProvider.java

```diff
@@ -0,0 +1,221 @@
+    List<AttachmentId> attachmentIds = identifierRelationController.listAttachmentIdsBySourceAndParentId(LinkedEventsConsts.IDENTIFIER_NAME, eventId);
+    List<Attachment> result = new ArrayList<>(attachmentIds.size());
+    
+    for (AttachmentId attachmentId : attachmentIds) {
+      Attachment attachment = linkedEventsAttachmentResourceContainer.get(attachmentId);
+      if (attachment != null) {
+        result.add(attachment);
+      }
+    }
+    
+    return result;
+  }
+  
+  @Override
+  public Attachment findEventImage(OrganizationId organizationId, EventId eventId, AttachmentId attachmentId) {
+    if (!identifierRelationController.isChildOf(eventId, attachmentId)) {
+      return null;
+    }
+    
+    return linkedEventsAttachmentResourceContainer.get(attachmentId);
+  }
+
+  @Override
+  public AttachmentData getEventImageData(OrganizationId organizationId, EventId eventId, AttachmentId attachmentId, Integer size) {
+    if (!identifierRelationController.isChildOf(eventId, attachmentId)) {
+      return null;
+    }
+    
+    AttachmentData imageData = getImageData(organizationId, attachmentId);
+    if (size != null) {
+      return scaleEventImage(imageData, size);
+    }
+    
+    return imageData;
+  }
+
+  private AttachmentData getImageData(OrganizationId organizationId, AttachmentId attachmentId) {
+    AttachmentData storedAttachmentData = getStoredAttachmentData(attachmentId);
+    if (storedAttachmentData != null) {
+      return storedAttachmentData;
+    }
+    
+    return downloadImageData(organizationId, attachmentId);
+  }
+  
+  private AttachmentData getStoredAttachmentData(AttachmentId attachmentId) {
+    StoredBinaryData storedBinaryData = linkedEventsAttachmentDataResourceContainer.get(attachmentId);
+    if (storedBinaryData != null) {
+      try {
+        return new AttachmentData(storedBinaryData.getContentType(), IOUtils.toByteArray(storedBinaryData.getDataStream()));
+      } catch (IOException e) {
+        logger.log(Level.SEVERE, "Failed to read stream data", e);
+      } 
+    }
+     
+    return null;
+  }
+
+  private boolean isWithinTimeRanges(Event event, OffsetDateTime startBefore,
+      OffsetDateTime startAfter, OffsetDateTime endBefore, OffsetDateTime endAfter) {
+    
+    OffsetDateTime eventStart = event.getStart();
+    OffsetDateTime eventEnd = event.getEnd();
+    
+    if (eventStart == null || eventEnd == null) {
+      return false;
+    }
+    
+    if (!isWithinTimeRange(startBefore, startAfter, eventStart)) {
+      return false;
+    }
+    
+    if (!isWithinTimeRange(endBefore, endAfter, eventEnd)) {
+      return false;
+    }
+    
+    return true;
+  }
+
+  private boolean isWithinTimeRange(OffsetDateTime before, OffsetDateTime after, OffsetDateTime time) {
+    if (before != null && time.isAfter(before)) {
+      return false;
+    }
+
+
+    if (after != null && time.isBefore(after)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)